test/unit/source/url/mihuashi_url_test.rb

```diff
@@ -32,8 +32,8 @@ class MihuashiUrlTest < ActiveSupport::TestCase
         )
 
         url_parser_should_work(
-          "https://www.mihuashi.com/profiles/29105?role=painter",
-          profile_url: "https://www.mihuashi.com/profiles/29105",
+          "https://www.mihuashi.com/profiles/29105",
+          profile_url: "https://www.mihuashi.com/profiles/29105?role=painter",
```

Comment:
I don't see the difference between https://www.mihuashi.com/profiles/29105 and https://www.mihuashi.com/profiles/29105?role=painter. Using `?role=painter` for the profile URL means all existing artist URLs will have to be normalized to that form (see <https://danbooru.donmai.us/artist_urls?search[url_matches]=*mihuashi*>).